src/turbolinks/controller.coffee

```diff
@@ -17,7 +17,10 @@ class Turbolinks.Controller
   start: ->
     unless @started
       addEventListener("click", @clickCaptured, true)
-      addEventListener("DOMContentLoaded", @pageLoaded, false)
+      if document.readyState is "loading"
+        addEventListener("DOMContentLoaded", @pageLoaded, false)
+      else
+        @pageLoaded()
```

Comment:
I think `@startHistory()` needs to be called before this condition so [`@location` is set](https://github.com/turbolinks/turbolinks/blob/master/src/turbolinks/controller.coffee#L59) for [`@pageLoaded()` to use](https://github.com/turbolinks/turbolinks/blob/master/src/turbolinks/controller.coffee#L138).